pkg/apis/crd/v1alpha2/types.go

```diff
@@ -543,3 +543,47 @@ type TrafficControlList struct {
 
 	Items []TrafficControl `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:validation:XValidation:rule="self.metadata.name == 'default'",message="Only one NodeLatencyMonitor with metadata.name=\"default\" can be created in the cluster"
+
+// NodeLatencyMonitor is used to monitor the latency between nodes in a Kubernetes cluster. It is a singleton resource,
+// meaning only one instance of it can exist in the cluster.
+type NodeLatencyMonitor struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec NodeLatencyMonitorSpec `json:"spec"`
+}
+
+type NodeLatencyMonitorSpec struct {
+	// +kubebuilder:validation:Pattern=`^[1-9][0-9]*[s]$`
+	// PingInterval specifies the interval between ping requests.
+	// Ping interval should be greater than or equal to 1s(one second).
+	// Defaults to "10s". Valid time units are "s".
+	PingInterval string `json:"pingInterval"`
+
+	// +kubebuilder:validation:Pattern=`^[1-9][0-9]*[s]$`
+	// PingTimeout specifies the timeout for each ping request.
+	// Ping timeout should be greater than or equal to 1s(one second).
+	// Defaults to "5s". Valid time units are "s".
+	PingTimeout string `json:"pingTimeout"`
+
+	// +kubebuilder:validation:Minimum=1
+	// PingConcurrentLimit specifies the maximum number of concurrent ping requests.
+	// Concurrency should be greater than or equal to 1.
+	// Defaults to 10. Valid range is unsigned integer.
+	PingConcurrentLimit int `json:"pingConcurrentLimit"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type NodeLatencyMonitorList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []NodeLatencyMonitor `json:"items"`
+}
```

Comment:
@antoninbas My previous implementation didn't add a List, but it seems to have failed to generate it in `make codegen`, suggesting that the type of List needs to be supplied.

Reply:
You can explicitly tell client-gen to only generate for get/watch etc., or to skip list https://gist.github.com/liangrog/1fc1cfbbfb0ef555729781c73caaef0b#file-k8s-code-gen-tags-L5

Reply:
If we don't add a list, we can't use the informer interface directly to listen to CRD resources, and if we use watch directly, we will probably handle the events manually in the same way as `cmd/antrea-agent-simulator/simulator.go`.

It may be necessary to get the crd resource via get for the first time when the node is online, and then listen to the resource changes via watch.

Reply:
I have added a comment for this `NodeLatencyMonitorList ` struct.